Source/Utils/WarpXAlgorithmSelection.H

```diff
@@ -11,6 +11,30 @@
 #include <AMReX_ParmParse.H>
 #include <string>
 
+/**
+  * \brief struct to determine the computational medium, i.e., vacuum or material/macroscopic
+           default is vacuum.
+  */
+struct MediumForEM {
+    enum {
+        Vacuum = 0,
+        Macroscopic = 1
+    };
+};
+
+/**
+  * \brief struct to select algorithm for macroscopic Maxwell solver
+           LaxWendroff represents sigma*E^(n+1/2) = sigma*0.5*(E^(n) + E^(n+1))
+           Backward Euler represents sigma*E^(n+1/2) = sigma*E^(n)
```

Comment:
Shouldn't this be `Backward Euler represents sigma*E^(n+1/2) = sigma*E^(n+1)`?

Reply:
If E is being evolved from E^{n-1/2} -> E^{n+1/2} I agree with dpgrote.  However I think it's less confusing if the LHS of these equations doesn't have any superscripts, e.g.,
`LaxWendroff represents sigma*E = sigma*0.5*(E^(n+1/2) + E^(n-1/2))`
`Backward Euler represents sigma*E = sigma*E^(n+1/2}`

Reply:
@dpgrote @ajnonaka E is updated from n-n+1. 

> Lax-Wendroff represents  sigma E = sigma*0.5*(E(n) + E(n+1))

and 

> Backward Euler represents sigma  E = sigma E(n)


The pdf attached in the PR has the derivation for E-update from E^(n-1/2) -> E^(n+1/2) to ensure that I get a form similar to that in Allen Taflove's book. In the implementation, I have E-update from E^(n)->E^(n+1).

I will modify the derivation in the pdf to reflect E-update from E^(n)->E^(n+1).

Reply:
Backward Euler: sigma*E = sigma*E(n+1)... you use the new-time.
A forward Euler scheme (not currently implemented) would use sigma*E = sigma*E(n).

Reply:
Sorry I will edit my above response. I have implemented sigma E = sigma E(n+1) and derived it that way too. But made a typo here. Thank you Andy for catching this. I will fix it in the comments as well.